src/VisualStudio/Core/Test/FindResults/FindResultsTests.vb

```diff
@@ -0,0 +1,179 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Text
+Imports System.Threading
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Editor.UnitTests
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Extensions
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Utilities
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.FindSymbols
+Imports Microsoft.VisualStudio.Composition
+Imports Microsoft.VisualStudio.LanguageServices.Implementation.Library.FindResults
+Imports Microsoft.VisualStudio.LanguageServices.UnitTests.ObjectBrowser.Mocks
+Imports Roslyn.Test.Utilities
+Imports Roslyn.Utilities
+
+Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.FindResults
+    Public Class FindResultsTests
+        <Fact, Trait(Traits.Feature, Traits.Features.FindReferences)>
+        <WorkItem(1138943)>
+        Public Sub ConstructorReferencesShouldNotAppearUnderClassNodeInCSharp()
+            Dim markup = <Text><![CDATA[
+class $$C
+{
+    const int z = 1;
+
+    public C() { }
+    public C(int x) { }
+
+    void T()
+    {
+        var a = new C();
+        var b = new C(5);
+        var c = C.z;
+    }
+}"]]></Text>
+
+            Dim expectedResults = New List(Of AbstractTreeItem) From
+                {
+                    New TestFindResult($"[CSharpAssembly1] C.C() ({ServicesVSResources.ReferenceCountSingular})",
+                        New TestFindResult("CSharpAssembly1\Test1.cs - (11, 21) : var a = new C();")),
+                    New TestFindResult($"[CSharpAssembly1] C.C(int) ({ServicesVSResources.ReferenceCountSingular})",
+                        New TestFindResult("CSharpAssembly1\Test1.cs - (12, 21) : var b = new C(5);")),
+                    New TestFindResult($"[CSharpAssembly1] class C ({ServicesVSResources.ReferenceCountSingular})",
+                        New TestFindResult("CSharpAssembly1\Test1.cs - (13, 17) : var c = C.z;"))
+                }
+
+            Verify(markup, LanguageNames.CSharp, expectedResults)
+        End Sub
+
+        <Fact, Trait(Traits.Feature, Traits.Features.FindReferences)>
+        <WorkItem(1138943)>
+        Public Sub ConstructorReferencesShouldNotAppearUnderClassNodeInVisualBasic()
+            Dim markup = <Text><![CDATA[
+Class C$$
+    Const z = 1
+
+    Public Sub New()
+    End Sub
+
+    Public Sub New(x As Integer)
+    End Sub
+
+    Sub T()
+        Dim a = New C()
+        Dim b = New C(5)
+        Dim d = C.z
+    End Sub
+End Class"]]></Text>
+
+            Dim expectedResults = New List(Of AbstractTreeItem) From
+                {
+                    New TestFindResult($"[VisualBasicAssembly1] Class C ({ServicesVSResources.ReferenceCountSingular})",
```

Comment:
Sadly, we don't control the ordering of top-level items in the actual display, so these are just in any order.